app/models/target.rb

```diff
@@ -27,4 +27,11 @@ class Target < ApplicationRecord
 
   validates :title, :longitude, :latitude, :radius, presence: true
   validates :radius, numericality: { only_integer: true }
+  validate :number_of_user_targets
+
+  private
+
+  def number_of_user_targets
+    errors.add(:user, 'exceeded number of targets limit') if user&.targets_limit_exceeded?
```

Comment:
WDYT about having the error message in a file to manage translations? Maybe using i18n or something

Reply:
Thanks for the comment, I will add this gem in another pr.